EXERCISE-1.md

```diff
@@ -195,7 +195,7 @@ each `stratum_bmv2` instance is generated in the
 
 ## 4. Program leaf1 using P4Runtime
 
-For this part we will use the [P4Runtime Shell](p4runtime-sh), an interactive
+For this part we will use the [P4Runtime Shell](https://github.com/p4lang/p4runtime-shell), an interactive
```

Comment:
This should be a reference-style link since the URL to the P4runtime Shell GitHub repo is defined at the end of this file as a Markdown reference. The issue here is the type of parentheses, it should be square, not round:
```suggestion
For this part we will use the [P4Runtime Shell][p4runtime-sh], an interactive
```